modules/nextflow/src/main/groovy/nextflow/scm/AssetManager.groovy

```diff
@@ -629,7 +629,7 @@ class AssetManager {
              */
             catch ( RefNotFoundException e ) {
                 def ref = checkoutRemoteBranch(revision)
-                return "checkout-out at ${ref.getObjectId().name()}"
+                return "checked out at ${ref?.getObjectId()?.name()}"
```

Comment:
Can `ref` be null? in this case the message reported would not be nice. Otherwise it can be kept the `ref.getObjectId().name()` version

Reply:
I think I added the null checks because at some point during testing it was null, which crashed the entire run with an even worse error message